library/modules/EventManager.cpp

```diff
@@ -200,6 +203,9 @@ static int32_t reportToRelevantUnitsTime = -1;
 //interaction
 static int32_t lastReportInteraction;
 
+//unit action
+static std::map<int32_t,std::vector<int32_t> > unitToKnownActions;
```

Comment:
Since we don't iterate through this structure, an `unordered_map` might bea better choice here.

Reply:
> Since we don't iterate through this structure, an `unordered_map` might bea better choice here.

True.

But an even better idea is to add a hash for the data we actually care about, since they are trivial to construct, and change it to `std::unordered_set<ActionData>`. This way we avoid whatever kind of search `std::find` is doing (I would presume linear, since it can't know if it's sorted.. but maybe that's a requirement else UB). It would also remove at least one indent level.